src/jit/lsra.cpp

```diff
@@ -4897,7 +4924,9 @@ LinearScan::allocateBusyReg(Interval *current, RefPosition *refPosition)
         if (assignedInterval == nullptr)
         {
             RefPosition* nextPhysRegPosition = physRegRecord->getNextRefPosition();
-            assert(nextPhysRegPosition->nodeLocation == refLocation && candidateBit != candidates);
+
+            // nodeLocation is not always == refLocation, so this assert is removed
+            //assert(nextPhysRegPosition->nodeLocation == refLocation && candidateBit != candidates);
             continue;
```

Comment:
I see that there is a possibility that it may require fixed registers AS WELL AS internal registers (i.e. if it is partially on the stack). That makes the suggested workaround more difficult.